src/testing/matchers/toRerenderWithSimilarSnapshot.ts

```diff
@@ -1,24 +1,36 @@
 import type {
   NextRenderOptions,
   RenderStream,
+  SnapshotStream,
 } from "@testing-library/react-render-stream";
 import { WaitForRenderTimeoutError } from "@testing-library/react-render-stream";
 import type { MatcherContext } from "expect";
 import type { MatcherFunction } from "expect";
 
-interface ToRerenderWithSimilarSnapshotOptions<T>
+import { getSerializableProperties } from "./utils/getSerializableProperties.js";
+
+export interface ToRerenderWithSimilarSnapshotOptions<T>
   extends Partial<NextRenderOptions> {
-  compare(previous: T, current: T): boolean;
+  expected?(previous: T): T;
 }
 
-export const toRerenderWithSimilarSnapshot: MatcherFunction<
-  [options: ToRerenderWithSimilarSnapshotOptions<any>]
-> = async function toRerenderWithSimilarSnapshot(
+export type ToEmitSimilarValueOptions<T> =
+  ToRerenderWithSimilarSnapshotOptions<T>;
+
+export type CommonStream<T = unknown> = {
+  getCurrent(): undefined | T;
+  takeNext(options: Partial<NextRenderOptions>): Promise<T>;
+};
+
+export const toEmitSimilarValue = async function toEmitSimilarValue(
   this: MatcherContext,
   actual,
-  options
+  {
+    expected: expect = (previous) => previous,
```

Comment:
```suggestion
    expected: getExpected = (previous) => previous,
```

Could we call this `getExpected` or even `getExpectedResult` (pick your favorite)? I saw `expect(...)` below and thought it was Jest's `expect` function. This would help avoid confusion.